src/OVAL/oval_component.c

```diff
@@ -1973,6 +2051,270 @@ static oval_syschar_collection_flag_t _oval_component_evaluate_ESCAPE_REGEX(oval
+				state = SLASH;
+			} else if (c == '?') {
+				// a ? matches any single character, but
+				// a ? at the begining of glob pattern can't match a .
+				// a ? never matches a / (see man 7 glob - Pathnames)
+				_string_append_string(regex, "[^./]");
+				state = NORMAL;
+			} else if (c == '*') {
+				// a * matches any string, but
+				// a * at the begining of glob pattern can't match a .
+				// a * never matches a / (see man 7 glob - Pathnames)
+				_string_append_string(regex, "(?=[^.])[^/]*");
+				state = NORMAL;
+			} else if (c == '.' || c == '|' || c == '^' || c == '(' || c == ')'
+					|| c == '{' || c == '}' || c == '+' || c == '$') {
+				_string_append_char(regex, '\\');
+				_string_append_char(regex, c);
+				state = NORMAL;
+			} else if (c == '[') {
+				_string_append_char(regex,'[');
+				state = LEFT_BRACKET;
+			} else if (c == '\\') {
+				if (noescape) {
+					_string_append_char(regex, '\\');
+					_string_append_char(regex, '\\');
+					state = NORMAL;
+				} else {
+					state = ESCAPE;
+				}
+			} else {
+				_string_append_char(regex, c);
+				state = NORMAL;
+			}
+			break;
+		case NORMAL:
+			if (c == '\0') {
+				goto finish;
+			} else if (c== '/') {
+				_string_append_char(regex, c);
+				state = SLASH;
+			} else if (c == '?') {
+				// a ? matches any single character, but
+				// it can never match a /
+				_string_append_string(regex, "[^/]");
+			} else if (c == '*') {
+				// a * matches any string, but
+				// it can never match a /
+				_string_append_string(regex, "[^/]*");
+			} else if (c == '.' || c == '|' || c == '^' || c == '(' || c == ')'
+					|| c == '{' || c == '}' || c == '+' || c == '$' ) {
+				_string_append_char(regex, '\\');
+				_string_append_char(regex, c);
+			} else if (c == '[') {
+				_string_append_char(regex,'[');
+				state = LEFT_BRACKET;
+			} else if (c == '\\') {
+				if (noescape) {
+					_string_append_char(regex, '\\');
+					_string_append_char(regex, '\\');
+				} else {
+					state = ESCAPE;
+				}
+			} else {
+				_string_append_char(regex, c);
+			}
+			break;
+		case LEFT_BRACKET:
+			if (c == '!') {
+				_string_append_char(regex, '^');
+			} else if (c == '\0') {
+				free(regex->str);
+				free(regex);
+				return NULL;
+			} else {
+				_string_append_char(regex, c);
+			}
+			state = CLASS;
+			break;
+		case CLASS:
+			if (c == '\\') {
+				if (noescape) {
+					_string_append_char(regex, '\\');
+				}
+			} else if (c == ']') {
+				state = NORMAL;
+			} else if (c == '\0') {
+				free(regex->str);
+				free(regex);
+				return NULL;
+			}
+			_string_append_char(regex, c);
+			break;
+		case ESCAPE:
+			// Only ?, *, [ and ] are escaped by a backslash
+			// Meaning of other characters is not affected by a backslash
+			if (c == '?' || c == '*' || c == '[' || c == ']') {
+				_string_append_char(regex, '\\');
+				_string_append_char(regex, c);
+			} else if (c == '\0') {
+				free(regex->str);
+				free(regex);
+				return NULL;
```

Comment:
I am not sure but I think this is not an error. We most likely just want to output the backslash in case \\ is the last character.